contracts/wormhole/contracts/mapping-token/darwinia/Sub2SubBacking.sol

```diff
@@ -0,0 +1,217 @@
+    event TokenRegisterFinished(bytes4 laneId, uint64 nonce, bool result);
+    event TokenUnlocked(address token, address recipient, uint256 amount);
+
+    modifier onlyOperatorOrOwner() {
+        require(operator == msg.sender || owner() == msg.sender, "Backing:caller is not the owner or operator");
+        _;
+    }
+
+    modifier onlySystem() {
+        require(SYSTEM_ACCOUNT == msg.sender, "System: caller is not the system account");
+        _;
+    }
+
+    function initialize() public initializer {
+        operator = msg.sender;
+        ownableConstructor();
+    }
+
+    function setMessagePalletIndex(uint32 index) external onlyOwner {
+        messagePalletIndex = index;
+    }
+
+    function changeDailyLimit(address token, uint amount) public onlyOwner  {
+        _changeDailyLimit(token, amount);
+    }
+
+    function unpause() external onlyOwner {
+        _unpause();
+    }
+
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    function updateOperator(address _operator) external onlyOperatorOrOwner {
+        operator = _operator;
+    }
+
+    function decimalConversion(uint256 balance) internal pure returns (uint256) {
+        return balance/(10**9);
+    }
+
+    /**
+     * @notice reigister new erc20 token to the bridge. Only owner or operator can do this.
+     * @param specVersion the spec_version of the bridged chain
+     * @param weight the remote dispatch call's weight
+     * @param laneId the laneId of the message channel
+     * @param token the original token address
+     * @param name the name of the original token
+     * @param symbol the symbol of the original token
+     * @param decimals the decimals of the original token
+     */
+    function registerErc20Token(
+        uint32 specVersion,
+        uint64 weight,
+        bytes4 laneId,
+        address token,
+        string memory name,
+        string memory symbol,
+        uint8 decimals
+    ) external payable onlyOperatorOrOwner {
+        require(registeredTokens[token] == false, "Backing:token has been registered");
+
+        // encode remote register dispatch call
+        bytes memory registerErc20 = SubToSubBridge(DISPATCH_ENCODER).encode_register_from_remote_dispatch_call(
+            specVersion,
+            weight,
+            ERC20_TOKEN_TYPE,
+            token,
+            name,
+            symbol,
+            decimals
+        );
+
+        // transform fee in contract(decimals is 18) to pallet(decimals is 9)
+        uint256 fee = decimalConversion(msg.value);
+
+        // encode 
+        // this fee is needed to send to the sub<>sub bridge fund to pay relayer
+        bytes memory sendMessageCall = SubToSubBridge(DISPATCH_ENCODER).encode_send_message_dispatch_call(
+            messagePalletIndex,
+            laneId,
+            registerErc20,
+            fee);
+
+        // send s2s message
+        (bool success, ) = DISPATCH.call(sendMessageCall);
```

Comment:
How the `fee` is deducted?

Reply:
This fee is payed in the pallet parity-bridges-common. We don't need to set fee in this call.

Reply:
It seems a force transfer?

Reply:
And can anyone encode the call to deduct the fee by call `DISPATCH.call(sendMessageCall)`?

Reply:
not force transfer, it's a inner not signed transfer https://github.com/paritytech/parity-bridges-common/blob/master/modules/messages/src/instant_payments.rs#L81

Reply:
ACK